modules/transformation.py

```diff
@@ -116,12 +116,11 @@ def initialize_status_info(file_name, page):
             "wptid": page.get("testID", base_name.split(".")[0]),
             "medianRun": 1,  # only available in RAW json (median.firstview.run), not HAR json
             "page": metadata.get("tested_url", ""),
-            "pageid": utils.clamp_integer(metadata["page_id"])
-            if metadata.get("page_id")
-            else None,
-            "rank": utils.clamp_integer(metadata["rank"])
-            if metadata.get("rank")
-            else None,
+            "pageid": utils.clamp_integer(
+                # `_metadata.pageid` used on and prior to 2022-03-01, `_metadata.page_id` used going forward
+                metadata["page_id"] if "page_id" in metadata else metadata["pageid"]
```

Comment:
Should we use the `get` method in case there is no `metadata`? Or do we want the AttributeError?

Not sure if we want a `None` value passed into `clamp_integer` though so this may need to be refactored slightly.

```suggestion
                metadata["page_id"] if "page_id" in metadata else metadata.get("pageid")
```

Reply:
In this case it will raise a `KeyError` since `page_id` and `pageid` are the only two valid keys and we should not be passing `None` downstream by using a `get()`. Currently, the code fails much farther downstream (i.e. when generating `requestid`) because of this.

Passing `None` to `clamp_integer()` will raise a `TypeError`, but that will not happen as implemented here since it will raise the `KeyError` first.

Reply:
Not sure I'm following. `metadata` was only recently added, so how do we get summary data for crawls before then? It seems like it will raise an exception and we won't get any summary data at all (or maybe the entire row is skipped?). I guess we could return `NULL` and subsequently run an update in BigQuery by JSONifying the entire `summary_pages` row.

Reply:
`log.pages[]._metadata` is available for HARs from March onward, however the key representing "page id" in the HARs changed between March and April from `pageid` to `page_id`. Since we do not care to load data prior to March into the new `all` tables, this code does not include any logic to handle the absence of metadata (i.e. an exception will be raised since metadata is expected). Summary data is generated without issues as well.

The PR description above contains links to successful March and April Dataflow jobs using this method. Here are links to a HAR from each month for reference as well as a sample from `all.pages` in BigQuery.

https://console.cloud.google.com/storage/browser/_details/httparchive/chrome-Mar_1_2022/220301_Dx0_K8.har.gz;tab=live_object?project=httparchive

https://console.cloud.google.com/storage/browser/_details/httparchive/crawls/chrome-Apr_1_2022/220401_Dx0_N2.har.gz;tab=live_object?project=httparchive

|client |date      |wptid        |metadata                                                                                  |summary                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
|-------|----------|-------------|------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|desktop|2022-04-01|220401_Dx0_N2|{"pageid":674,"crawlid":659,"rank":1000,"layout":"Desktop","location":"California:Chrome"}|<details><summary>EXPAND</summary><p>{"metadata": "{\"pageid\": 674, \"crawlid\": 659, \"rank\": 1000, \"layout\": \"Desktop\", \"location\": \"California:Chrome\"}", "pageid": 674, "createDate": 1661725904, "startedDateTime": 1648788132, "archive": "All", "label": "Apr 1 2022", "crawlid": 659, "url": "", "urlhash": 54301, "urlShort": "", "TTFB": 635, "renderStart": 700, "fullyLoaded": 782, "visualComplete": 700, "onLoad": 711, "gzipTotal": 0, "gzipSavings": 0, "numDomElements": 11, "onContentLoaded": 674, "cdn": "Akamai", "SpeedIndex": 700, "PageSpeed": null, "_connections": 1, "_adult_site": false, "avg_dom_depth": 3, "doctype": null, "document_height": 695, "document_width": 1366, "localstorage_size": 0, "sessionstorage_size": 0, "meta_viewport": null, "num_iframes": 0, "num_scripts": 0, "num_scripts_sync": 0, "num_scripts_async": 0, "usertiming": 0, "reqTotal": 3, "bytesTotal": 1944, "reqJS": 0, "bytesJS": 0, "reqImg": 2, "bytesImg": 1296, "reqJson": 0, "bytesJson": 0, "reqCss": 0, "bytesCss": 0, "reqHtml": 1, "bytesHtml": 648, "reqFont": 0, "bytesFont": 0, "reqOther": 0, "bytesOther": 0, "reqAudio": 0, "bytesAudio": 0, "reqVideo": 0, "bytesVideo": 0, "reqText": 0, "bytesText": 0, "reqXml": 0, "bytesXml": 0, "reqGif": 0, "bytesGif": 0, "reqJpg": 1, "bytesJpg": 648, "reqPng": 0, "bytesPng": 0, "reqWebp": 0, "bytesWebp": 0, "reqSvg": 0, "bytesSvg": 0, "reqFlash": 0, "bytesFlash": 0, "numDomains": 1, "maxageNull": 3, "maxage0": 0, "maxage1": 0, "maxage30": 0, "maxage365": 0, "maxageMore": 0, "bytesHtmlDoc": 648, "numRedirects": 0, "numErrors": 0, "numGlibs": 0, "numHttps": 3, "numCompressed": 0, "maxDomainReqs": 2, "wptid": "220401_Dx0_N2", "wptrun": 1, "rank": 1000}</p></summary>            |
|desktop|2022-03-01|220301_Dx0_K8|{"pageid":"616","crawlid":"657","rank":"1000"}                                            |<details><summary>EXPAND</summary><p>{"metadata": "{\"pageid\": \"616\", \"crawlid\": \"657\", \"rank\": \"1000\"}", "pageid": 616, "createDate": 1661579464, "startedDateTime": 1646113224, "archive": "All", "label": "Mar 1 2022", "crawlid": "657", "url": "", "urlhash": 54301, "urlShort": "", "TTFB": 523, "renderStart": 1300, "fullyLoaded": 3807, "visualComplete": 3700, "onLoad": 3550, "gzipTotal": 359326, "gzipSavings": 52588, "numDomElements": 407, "onContentLoaded": 1239, "cdn": "Facebook", "SpeedIndex": 2117, "PageSpeed": null, "_connections": 5, "_adult_site": false, "avg_dom_depth": 9, "doctype": "html", "document_height": 970, "document_width": 1351, "localstorage_size": 27, "sessionstorage_size": 235, "meta_viewport": null, "num_iframes": 0, "num_scripts": 32, "num_scripts_sync": 1, "num_scripts_async": 16, "usertiming": 0, "reqTotal": 35, "bytesTotal": 1530884, "reqJS": 18, "bytesJS": 244868, "reqImg": 7, "bytesImg": 1151120, "reqJson": 0, "bytesJson": 0, "reqCss": 5, "bytesCss": 16399, "reqHtml": 2, "bytesHtml": 24822, "reqFont": 3, "bytesFont": 93675, "reqOther": 0, "bytesOther": 0, "reqAudio": 0, "bytesAudio": 0, "reqVideo": 0, "bytesVideo": 0, "reqText": 0, "bytesText": 0, "reqXml": 0, "bytesXml": 0, "reqGif": 0, "bytesGif": 0, "reqJpg": 1, "bytesJpg": 9082, "reqPng": 5, "bytesPng": 1072344, "reqWebp": 0, "bytesWebp": 0, "reqSvg": 0, "bytesSvg": 0, "reqFlash": 0, "bytesFlash": 0, "numDomains": 4, "maxageNull": 3, "maxage0": 0, "maxage1": 0, "maxage30": 4, "maxage365": 28, "maxageMore": 0, "bytesHtmlDoc": 24822, "numRedirects": 0, "numErrors": 0, "numGlibs": 0, "numHttps": 35, "numCompressed": 0, "maxDomainReqs": 27, "wptid": "220301_Dx0_K8", "wptrun": 1, "rank": 1000}</p></details>|

Reply:
Since this PR is only meant to cover the ability to backfill from March 2022 onward, I'll open an issue to cover even older datasets.